frontend/talentsearch/src/js/components/pool/PoolAdvertisementPage.tsx

```diff
@@ -131,8 +134,19 @@ const PoolAdvertisement = ({
   const genericTitle = classification?.genericJobTitles?.length
     ? classification.genericJobTitles[0]
     : null;
-  const classificationSuffix = `${classification?.group}-0${classification?.level}`;
-  const fullTitle = getFullPoolAdvertisementTitle(intl, poolAdvertisement);
```

Comment:
We should probably consolidate the two functions `formattedPoolPosterTitle` and `getFullPoolAdvertisementTitle`, since they are both meant to return the same string. Also, `getFullPoolAdvertisementTitle` is used across the "Apply to a pool" process so we need to replace those as well.

I like that `getFullPoolAdvertisementTitle` does the heavy lifting and only requires the `poolAdvertisement`, so I'm thinking bring the logic you wrote on this file to that function. Then let it live in the `poolUtils.ts` helper file in common. The function would be something like this?

```ts
import { getPoolStream } from "@common/constants/localizedConstants";
import { getLocalizedName } from "@common/helpers/localize";
import { formatClassificationString } from "@common/helpers/poolUtils";
import type { IntlShape } from "react-intl";
import { Maybe, PoolAdvertisement } from "../../api/generated";

const getFullPoolAdvertisementTitle = (
  intl: IntlShape,
  advertisement: Maybe<PoolAdvertisement>,
): string => {
  if (advertisement === null || advertisement === undefined)
    return intl.formatMessage({
      id: "D91nGW",
      defaultMessage: "Job title not found.",
      description:
        "Message shown to user when pool name or classification are not found.",
    });

  const classification = advertisement.classifications
    ? advertisement.classifications[0]
    : null;

  let classificationSuffix = ""; // type wrangling the complex type into a string
  if (classification) {
    classificationSuffix = formatClassificationString({
      group: classification?.group,
      level: classification?.level,
    });
  }
  const genericTitle = getLocalizedName(advertisement.name, intl);

  return `${genericTitle ? `${genericTitle} ` : ""}(${classificationSuffix}${
    advertisement.stream
      ? ` ${intl.formatMessage(getPoolStream(advertisement.stream))}`
      : ""
  })`;
};

export default getFullPoolAdvertisementTitle;
```

Reply:
Good call. I wasn't sure if they were supposed to return the same thing or not.

Reply:
Both functions have now been consolidated into a new `getFullPoolAdvertisementTitle` function located in `poolUtils.ts`.